app-devel/adaptivecpp/autobuild/defines

```diff
@@ -0,0 +1,17 @@
+PKGNAME=adaptivecpp
+PKGSEC=devel
+PKGDEP="boost gcc-runtime glibc llvm python-3"
+PKGDES="Independent, community-driven compiler for C++-based heterogeneous programming models"
+ABTYPE=cmake
+BUILDDEP="boost llvm python-3"
+
+CMAKE_AFTER=" \
+    -DACPP_CONFIG_FILE_GLOBAL_INSTALLATION=ON \
+    -DDEFAULT_TARGETS='omp.accelerated' \
+    -DWITH_ACCELERATED_CPU=ON \
+    -DWITH_CUDA_BACKEND=OFF \
+    -DWITH_LEVEL_ZERO_BACKEND=OFF \
+    -DWITH_OPENCL_BACKEND=OFF \
```

Comment:
Why disable all these features?

Reply:
Similar to LLVM's OpenMP GPU offloading capabilities, all these features require the availability of corresponding components (such as CUDA, ROCM, and OpenCL implementations) both at compile time and runtime. However, we cannot guarantee that these components will always be available or that they will match between the compile and runtime environments. The only component consistently available is LLVM OpenMP on the host. Users have to build other features from source on their own machines.

Additionally, SSCP operates on a JIT compilation flow, allowing a single binary to potentially function across multiple backends like OpenCL and CUDA without recompilation. However, we can only support one host backend due to the constraints, and no backend switching would occur. Therefore, it is better to utilize AOT compilation over JIT, which has more overhead.